### ISSUE TITLE
trigger `remove` event before `stopListening()` in `remove` method

### ISSUE THREAD

**Issue by mmacaula** [AmpersandJS/ampersand-view#120]
I have a view that needs some cleanup to happen before closing, I look and see a handy `remove` event for when the view is removed, so i do this:

```
View.extend({
  initialize : function(){
    this.listenTo(this,'remove',this.cleanup);
  },
  cleanup : function(){ 
  // do cleanup
  }
});

```
unfortunately, this doesn't work because during the remove call [stopListening](https://github.com/AmpersandJS/ampersand-view/blob/master/ampersand-view.js#L150)  is called which prevents `cleanup` from getting called.  I can go around this by using `this.once`  or `this.on` instead of `listenTo` and that works ok, but seems like I need to know a lot more about a view lifecycle and the details of how remove works than I should.  And I bet that this has tripped up other users as well. 

I propose moving the trigger call up a couple of lines to before `stopListening` is called.  Then it will work for all cases and as (I think) expected.

**Comment by aaronmccall** [comment]
@mmacaula, what about just overriding `remove`?

```javascript
View.extend({
  remove: function () {
    this.cleanup();
    View.prototype.remove.call(this);
  }
  cleanup : function(){ 
  // do cleanup
  }
});
```

**Comment by mmacaula** [comment]
@aaronmccall yeah that will work too, as will the other methods I mentioned above.  I'm just thinking its non-intuitive, and a source of an easily avoided mistake.  Actually that's why I created this issue.  We just ran into this problem and it was enough of a headache that I thought it shouldn't be that way.  We have this nice 'remove' event but the safest way to register to listen to it (via listenTo) doesn't work.  I need to take a look to see if my proposed change breaks any tests but I'm hopeful it won't.

**Comment by lukekarrys** [comment]
Thanks for the issue @mmacaula. I think you're right, it doesn't make much sense to trigger an event after calling `stopListening()`. Would you be able to work up a PR with a test? I think it would be enough to just put the `remove` event trigger right before `stopListening()`

**Comment by mmacaula** [comment]
Sure I'd love to

**Comment by lukekarrys** [comment]
Thanks @mmacaula!

**Comment by lukekarrys** [comment]
Closed by #122
